packages/components/src/accordion/use-accordion.ts

```diff
@@ -3,47 +3,48 @@ import { mergeRefs } from "@chakra-ui/hooks/use-merge-refs"
 import { callAllHandlers } from "@chakra-ui/utils/call-all"
 import { createContext } from "@chakra-ui/utils/context"
 import { warn } from "@chakra-ui/utils/warn"
-import { useCallback, useEffect, useId, useRef, useState } from "react"
-import {
-  useAccordionDescendant,
-  useAccordionDescendants,
-} from "./accordion-context"
+import { nextById, prevById, queryAll } from "@zag-js/dom-utils"
+import { useCallback, useId, useRef, useState } from "react"
+
+export type AccordionValue<T extends boolean | undefined> = T extends undefined
+  ? string
+  : T extends true
+  ? string[]
+  : string
 
 /* -------------------------------------------------------------------------------------------------
  * useAccordion - The root react hook that manages all accordion items
  * -----------------------------------------------------------------------------------------------*/
 
-/**
- * @deprecated - This will be removed in future versions.
- * Please use `number | number[]` instead.
- */
-export type ExpandedIndex = number | number[]
-
-export interface UseAccordionProps {
+export interface UseAccordionProps<Multiple extends boolean> {
   /**
    * If `true`, multiple accordion items can be expanded at once.
    *
    * @default false
    */
-  allowMultiple?: boolean
+  allowMultiple?: Multiple
```

Comment:
We should keep this as a boolean because we want the value always to return an array (regardless of whether it's  multiple or not)

Reply:
I see Ark uses an array as well, will update it.